projects/packages/assets/src/class-assets.php

```diff
@@ -349,8 +349,9 @@ public static function register_script( $handle, $path, $relative_to, array $opt
 			'strategy'         => '',
 			'textdomain'       => null,
 		);
+		'@phan-var array{asset_path:?string,async:bool,css_dependencies:string[],css_path:?string,dependencies:string[],enqueue:bool,in_footer:bool,media:string,minify:?bool,nonmin_path?:string,strategy:string,textdomain:string,version?:string} $options'; // Phan gets confused by the array addition.
 
-		if ( $options['css_path'] && substr( $options['css_path'], -4 ) !== '.css' ) {
+		if ( is_string( $options['css_path'] ) && $options['css_path'] !== '' && substr( $options['css_path'], -4 ) !== '.css' ) {
```

Comment:
We still want to throw an error if `css_path` is an empty string, right? As written, the current change would continue with an empty string.

Given we're just wanting to make sure that the string ends in `.css`, I'd go with this (thanks to WP polyfills):

```suggestion
		if ( is_string( $options['css_path'] ) && ! str_ends_with( $options['css_path'], '.css' ) ) {
```

Or we could even cast a wider net if we really wanted:
```suggestion
		if ( ! is_string( $options['css_path'] ) ) {
			throw new \InvalidArgumentException( '$options[\'css_path\'] must be a string!' );
		} elseif ( ! str_ends_with( $options['css_path'], '.css' ) ) {
```

Reply:
I'm not sure the `str_ends_with` polyfill is guaranteed to be loaded everywhere where this might be called.

Of the documented inputs (null|string), the old logic didn't throw for null or empty string or string-0. The new logic continues to not throw for null or empty-string, but will for string-0.

As for undocumented inputs, the new logic ignores any non-strings while the old logic might have ignored (false or 0), stringified, or thrown a fatal.

Reply:
> the old logic didn't throw for [...] empty string

Ah, good call. I was misreading the first condition.